scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.ApplePayEventReceiver.cs.erb

```diff
@@ -80,16 +149,67 @@ namespace <%= namespace %>.SDK.iOS {
             }
         }
 
+         private ApplePayAuthorizationStatus GetAuthorizationStatusFromCheckoutUserErrors(List<UserError> errors) {
+
+            ApplePayAuthorizationStatus statusToReturn = ApplePayAuthorizationStatus.Failure;
+
+            foreach (var error in errors) {
+                var fields = error.field();
+
+                // If there exists a UserErrorPath that is atleast 2 Fields long, and the second last field is `billingAddress`
+                if (fields.Count > 2 && fields[fields.Count - 2].Equals("billingAddress")) {
+                    statusToReturn = ApplePayAuthorizationStatus.InvalidBillingPostalAddress;
+                } else {
+                    statusToReturn = ApplePayAuthorizationStatus.Failure;
+                    break;
+                }
+            }
+
+            return statusToReturn;
+        }
+
+        private ApplePayAuthorizationStatus GetAuthorizationStatusFromShippingUserErrors(List<UserError> errors) {
+
+            string[] shippingAddressFields = {"address1", "address2", "city", "country", "province", "zip"};
+            var shippingAddressFieldsSet = new HashSet<string>(shippingAddressFields);
+
+            string[] shippingContactFields = {"firstName", "lastName", "phone", "email"};
+            var shippingContactFieldsSet = new HashSet<string>(shippingContactFields);
+
+            ApplePayAuthorizationStatus statusToReturn = ApplePayAuthorizationStatus.Failure;
+
+            foreach (var error in errors) {
+                var fields = error.field();
+                var lastField = fields[fields.Count - 1];
```

Comment:
I talked with Mikko, these checks should happen higher up, when we construct a `ShopifyError` because this would be an issue with the SDK